spec/support/spree/config_helper.rb

```diff
@@ -0,0 +1,18 @@
+module Spree
+  module ConfigHelper
+    def with_spree_config(hash)
+      stored = config_slice(hash.keys)
+      Config.set(hash)
+      yield
+      Config.set(stored)
```

Comment:
My understanding is that `.set` just changes the value of the settings corresponding to the keys provided. So I think the code already does what you want it to do. At least, that was my intention. Perhaps I need some specs...